docarray/base_doc/doc.py

```diff
@@ -207,73 +332,193 @@ def _docarray_to_json_compatible(self) -> Dict:
+                # we need to do this because pydantic will not recognize DocList correctly
+                original_exclude = original_exclude or {}
+                if field not in original_exclude:
+                    data[field] = getattr(
+                        self, field
+                    )  # here we need to keep doclist as doclist otherwise if a user want to have a special json config it will not work
+
+            # this is copy from pydantic code
+            if self.__custom_root_type__:
+                data = data[ROOT_KEY]
+            return self.__config__.json_dumps(data, default=encoder, **dumps_kwargs)
+
+        def dict(
+            self,
+            *,
+            include: Optional[Union['AbstractSetIntStr', 'MappingIntStrAny']] = None,
+            exclude: ExcludeType = None,
+            by_alias: bool = False,
+            skip_defaults: Optional[bool] = None,
+            exclude_unset: bool = False,
+            exclude_defaults: bool = False,
+            exclude_none: bool = False,
+        ) -> 'DictStrAny':
+            """
+            Generate a dictionary representation of the model, optionally specifying
+            which fields to include or exclude.
+
+            """
+
+            exclude, original_exclude, doclist_exclude_fields = self._exclude_doclist(
+                exclude=exclude
+            )
+
+            data = super().dict(
                 include=include,
                 exclude=exclude,
+                by_alias=by_alias,
+                skip_defaults=skip_defaults,
                 exclude_unset=exclude_unset,
                 exclude_defaults=exclude_defaults,
                 exclude_none=exclude_none,
             )
-        )
 
-        # this is the custom part to deal with DocList
-        for field in doclist_exclude_fields:
-            # we need to do this because pydantic will not recognize DocList correctly
-            original_exclude = original_exclude or {}
-            if field not in original_exclude:
-                data[field] = getattr(
-                    self, field
-                )  # here we need to keep doclist as doclist otherwise if a user want to have a special json config it will not work
-
-        # this is copy from pydantic code
-        if self.__custom_root_type__:
-            data = data[ROOT_KEY]
-        return self.__config__.json_dumps(data, default=encoder, **dumps_kwargs)
+            for field in doclist_exclude_fields:
+                # we need to do this because pydantic will not recognize DocList correctly
+                original_exclude = original_exclude or {}
+                if field not in original_exclude:
+                    val = getattr(self, field)
+                    data[field] = (
+                        [doc.dict() for doc in val] if val is not None else None
+                    )
+
+            return data
+
+    else:
+
+        def model_dump(  # type: ignore
```

Comment:
Am I understanding correctly that in v2 json and dict are both subsumed by this model_dump? So if the user wants json they would to `model_dump()` first and then convert that to json themselves?

Reply:
model_dump is used under the hood to both seriliaze to json and to "dict". Basically now in pydantic v2 calling .dict() and .json() will call under the hood this model_dump